### PR TITLE
Releases: Variablize Release Notes

### PR DESCRIPTION
**Pull request by harding**
HTML site diff (English only): http://dg2.dtrt.org/diff.html

This is a re-do of pull #751 that only implements the parts @saivann and I could both agree on per [this comment](https://github.com/bitcoin/bitcoin.org/pull/751#issuecomment-74931608).

It adds additional variables to the release note files to allow setting the version number and date.  The version number is required, and can be used to automatically set the release notes title.  The date is optional and can be set hours/days after the release.

Additionally, a Makefile test is added that checks whether the download files exist on the Bitcoin.org server.  This can help prevent creating a broken Download page.

Changes in commit 404af8bfdf0825ee98f892301315db1b5aa66dcc

* Set variables for all previous releases

* Document variables in README.md

* Update code and templates to use variables

* Add the Download page links to the "dl" CSS class. Also add newlines to make the HTML a bit easier to parse using sed

* Add a new Liquid plugin to print warnings. This is used to print a non-error warning if any release is created without the optional date

Changes in commit b88f6ee59c26e8bfd1f938c23a9154eb5d03e7d8

* Renamed the old release files:

        cd _releases
        rename 's/^.*-v//' *
        git add *

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -272,21 +272,55 @@ issue][] if a Bitcoin meetup event isn't displayed.)
...
+in and replacing information as necessary, and then reformatting the
+release notes (if necessary) as described by the last lines of the YAML
+header.
 
-...
-```
-* `VERSION` is used to define the version and construct the URL. It should adhere to this format: `v0.3.24`.
-* `title: ...` will be used as the title.
-* `magnetlink: ...` (optional) the urlencoded torrent magnet link on the download page.
+Download links will automatical be set to the defaults using the current
```
Comment (saivann) [inline README.md]: @harding s/automatical/automatically/

**Comment by saivann** [comment]
@harding LGTM! I've quickly tested the result and reviewed the code.

**Comment by harding** [comment]
@saivann thanks!  (I'll fix that typo.)

In the absence of critical feedback, this will be merged around 01:00 UTC Friday.

**Comment by harding** [comment]
Had to rebase to fix Makefile conflict with #776, so merged as https://github.com/bitcoin/bitcoin.org/commit/18defab52c79d355c80b1bc949f85b8e572bfe9a

**Pull request closed without merging**